### PR TITLE
feat(commerce): add search analytics

### PR DESCRIPTION
**Pull request by Spuffynism**
Add commerce search analytics events.

~⚠️ I'm not sure what I should be adding for the `analyticsType` and `analyticsPayloadBuilder` properties of the search actions. 🤔~ Edit: We'll tackle this in CAPI-325 and CAPI-326

[CAPI-244]

[CAPI-244]: https://coveord.atlassian.net/browse/CAPI-244?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/headless/src/features/commerce/search/search-analytics-actions.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import {OmniboxSuggestionMetadata} from '../../query-suggest/query-suggest-analytics-actions';
+
+export const logInterfaceLoad = (): CommerceSearchAction =>
+  makeCommerceAnalyticsAction({
+    prefix: 'analytics/commerce/interface/load',
+    __legacy__getBuilder: (client) => client.makeInterfaceLoad(),
+    analyticsType: '',
+    analyticsPayloadBuilder: (_state) => {
+      return {};
+    },
```
Comment (Spuffynism) [inline packages/headless/src/features/commerce/search/search-analytics-actions.ts:14]: What goes in these? Looking at the [analytics schema](https://github.com/coveo/analytics_schema/tree/master/schemas/client), I don't see a "commerce search" event.

Comment (fbeaudoincoveo) [reply packages/headless/src/features/commerce/search/search-analytics-actions.ts:14]: I'm not sure: is there a plan to define one, or are we using the [regular search schemas](https://github.com/coveo/analytics_schema/tree/master/schemas/internal/api.search.search/2)?

@pvanool , thoughts?

Comment (pvanool) [reply packages/headless/src/features/commerce/search/search-analytics-actions.ts:14]: It's currently on pause. We definitely want to go forward with it in the future. Is this something that we can easily add when we have a schema ready?

Comment (Spuffynism) [reply packages/headless/src/features/commerce/search/search-analytics-actions.ts]: As stated in the PR description, we'll tackle this in CAPI-325 and CAPI-326

**Review by louis-bompart — CHANGES_REQUESTED** [review]
The events are sent through the old stack, that ain't gonna fly, even for the purpose of the demo.
We can build subsequent PRs on top, but for Search CMH, I don't think we should merge that, as, afaik, search reported this way won't wwork down the line (i.e in the CMH UI).

**Comment by Spuffynism** [comment]
@louis-bompart Ok, closing this until we figure out how analytics should work for headless commerce. https://coveord.atlassian.net/browse/CAPI-325 and https://coveord.atlassian.net/browse/CAPI-326 will cover this follow-up work.

**Pull request closed without merging**